### PR TITLE
Sort person command

### LINKED ISSUE #114: As a user with many contacts, I want to sort the contacts by name
so that I can view the contacts in alphabetical order

### PR DESCRIPTION
**Pull request by siqiang-ng**
Fixes #114

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -157,6 +157,23 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 ## **Implementation**
 
 ### Implemented features
+#### Contact features
+**SortContact feature**
+The `sort` command (class SortContactCommand) is a SortContact feature which allows a user to sort the current person list in alphabetical order permanently. Contacts that are being added to the list later will not be sorted and added to the end of the list.
```
Comment (yeo-kexin) [inline docs/DeveloperGuide.md]: i think this part is a bit confusing especially the command and features part? 
(also what is the definition of feature vs command? because I've always interpreted feature as the whole aspect (e.g. contacts, tags, tasks...), and commands to be like what you can do with those features but maybe I'm wrong?)

also, i think the command name is normally named after the command word? so this could be kind of confusing

Comment (yeo-kexin) [inline docs/DeveloperGuide.md]: should it be able to go back to chronological order? else, it feels like this should be our default setting since there's no other option

File: src/main/java/seedu/address/logic/commands/contactcommands/SortContactCommand.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package seedu.address.logic.commands.contactcommands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.person.PersonNameComparator;
+
+public class SortContactCommand extends Command {
```
Comment (yeo-kexin) [inline src/main/java/seedu/address/logic/commands/contactcommands/SortContactCommand.java]: previously raised, but maybe it should be SortCommand? or we change everything to fit e.g. AddContact...

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -194,7 +194,7 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleExit();
             }
 
-            if (commandResult.isTagList()) { // TODO: Remove this in v1.3
```
Comment (yeo-kexin) [inline src/main/java/seedu/address/ui/MainWindow.java]: haha wait why did we need to remove this in the first place? i remember it was related to some hack thing?

**Review by yeo-kexin — COMMENTED** [review]
some areas that may need to be edited?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (siqiang-ng) [reply docs/DeveloperGuide.md]: No apparently I tried to think of ways to do it but we could not because of the fact that UI get the Filtered List instead of the Sorted List, unless we implement the undo feature?

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (siqiang-ng) [reply src/main/java/seedu/address/ui/MainWindow.java]: I remove the comment only

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (siqiang-ng) [reply docs/DeveloperGuide.md]: hmmm if like this, then we might have to change all to command only

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (yeo-kexin) [reply src/main/java/seedu/address/ui/MainWindow.java]: haha sorry let me rephrase, why was that comment there in the first place?

**Review by akgrenSoar — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (siqiang-ng) [reply src/main/java/seedu/address/ui/MainWindow.java]: Ohh erm I was really confused w the implementation initially, because I think this is not a good long-term solution but I guess for now, we should just proceed

**Review by yeo-kexin — APPROVED** [review]
LGTM!
